lightningd/notification.c

```diff
@@ -122,29 +123,35 @@ void notify_disconnect(struct lightningd *ld, struct node_id *nodeid)
 
 /*'warning' is based on LOG_UNUSUAL/LOG_BROKEN level log
  *(in plugin module, they're 'warn'/'error' level). */
-static void warning_notification_serialize(struct json_stream *stream,
+static void log_notification_serialize(struct json_stream *stream,
 					   struct log_entry *l)
 {
-	json_object_start(stream, "warning");
-	/* Choose "BROKEN"/"UNUSUAL" to keep consistent with the habit
-	 * of plugin. But this may confuses the users who want to 'getlog'
-	 * with the level indicated by notifications. It is the duty of a
-	 * plugin to eliminate this misunderstanding.
-	 */
-	json_add_string(stream, "level",
-			l->level == LOG_BROKEN ? "error"
-			: "warn");
+	if (l->level >= LOG_UNUSUAL)
+		json_object_start(stream, "warning");
+	else
+		json_object_start(stream, "log");
```

Comment:
Seems very strange to me that we are partitioning the `log` topic into `warning` and `log`. I would expect `log` to contain all the log entries that have been emitted.

Reply:
I did this for retro compatibility, but I did not use the deprecated API, maybe I should use it

Reply:
OK, I have restored the previous warning log as it appears my changes were unnecessarily disruptive. The warning notification serves a different purpose from the log notification.

Thank you for pointing this out.